src/stdatamodels/jwst/transforms/models.py

```diff
@@ -672,34 +737,36 @@ def apply_poly(coeff_model, inputs, t):
                 else:
                     sumval = coeff_model(t)
             else:
-                sumval = 0.
+                sumval = 0.0
                 for i in range(ord_t):
-                    sumval += t ** i * coeff_model[i](*inputs[:coeff_model[i].n_inputs])
+                    sumval += t**i * coeff_model[i](*inputs[: coeff_model[i].n_inputs])
             return sumval
 
         l_poly = apply_poly(lmodel, (x0, y0), t)
 
         return x0, y0, l_poly, order
 
     def invdisp_interp(self, order, x0, y0, dx):
-
         if len(dx.shape) == 2:
             dx = dx[0, :]
 
         t_len = dx.shape[0]
-        t0 = np.linspace(0., 1., t_len)
+        t0 = np.linspace(0.0, 1.0, t_len)
 
         if isinstance(self.xmodels[order], (ListNode, list)):
             if len(self.xmodels[order]) == 2:
                 xr = self.xmodels[order][0](x0, y0) + t0 * self.xmodels[order][1](x0, y0)
             elif len(self.xmodels[order]) == 3:
-                xr = self.xmodels[order][0](x0, y0) + t0 * self.xmodels[order][1](x0, y0) + \
-                     t0**2 * self.xmodels[order][2](x0, y0)
+                xr = (
+                    self.xmodels[order][0](x0, y0)
+                    + t0 * self.xmodels[order][1](x0, y0)
+                    + t0**2 * self.xmodels[order][2](x0, y0)
+                )
             elif len(self.xmodels[order][0].inputs) == 1:
                 xr = (dx - self.xmodels[order][0].c0.value) / self.xmodels[order][0].c1.value
                 return xr
             else:
-                raise Exception
+                raise Exception  # noqa: TRY002
```

Comment:
Can this be made more specific instead of bare Exception?

Reply:
Similar to https://github.com/spacetelescope/stdatamodels/pull/383#discussion_r1937770506 this shows no test coverage. We could add a new exception class and use it here but I'd like to see tests first.

Reply:
This feels to me like a pretty safe one, since any exception would inherit from `Except`.

Reply:
What do you suggest we name the new exception and add to it's docstring?

Reply:
Maybe something like `InvalidTransformError(f"xmodels for order {order} has invalid length {len(self.xmodels)}")`

But I see your point that we might want to run that past, say, Nadia or someone more expert with the code to ensure that name and docstring seem correct to her... and in that case it would make more sense as a separate PR.

Reply:
Since this is also covered by the issue you just opened, I'd say we leave it as-is here and ignore my original comment.  Feel free to resolve if you agree

Reply:
Thanks for the suggestion. I opened https://github.com/spacetelescope/stdatamodels/issues/389 to track this.